internal/cmd/login.go

```diff
@@ -113,3 +117,36 @@ func pingAPI(ctx context.Context, workspaceURL *url.URL, token string) error {
 	}
 	return nil
 }
+
+// isWSL determines if coder-cli is running within Windows Subsystem for Linux
+func isWSL() (bool, error) {
+	if runtime.GOOS == "darwin" || runtime.GOOS == "windows" {
+		return false, nil
+	}
+	data, err := ioutil.ReadFile("/proc/version")
+	if err != nil {
+		return false, xerrors.Errorf("read /proc/version: %w", err)
+	}
+	return strings.Contains(strings.ToLower(string(data)), "microsoft"), nil
+}
+
+// openURL opens the provided URL via user's default browser
+func openURL(url string) error {
+	var cmd string
+	var args []string
+
+	wsl, err := isWSL()
```

Comment:
I think this is good, but I do wonder whether we could propose this upstream to pkg/browser

Reply:
Thank you for the review. I agree, this work should ideally be  part of upstream `pkg/browser`. But Thoughts on moving forward with the work for Coder until I have the patch ready for upstream?

Reply:
Yeah, I think it makes sense to make this change now for sure! No objection to it :)

Thanks for your contribution!